### PR TITLE
feat(alerting): make authentication optional for email provider

### PR DESCRIPTION
**Pull request by calvinhenderson**
## Summary

#607 

## Checklist

- [X] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [X] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

**Comment by calvinhenderson** [comment]
I noticed an issue where emails are rejected when using a non-local smtp server due to the _from hostname_ being set to `localhost`. I updated the `gomail.Dialer` initialization to set the `LocalName` attribute to the `From` domain when disabling auth. This resolved the issue and email alerts are now working correctly utilizing the Gmail SMTP relay.

File: README.md [hunk 1]
```diff
@@ -468,20 +468,21 @@ endpoints:
...
+| `alerting.email.from`                   | Email used to send the alert                                                               | Required `""` |
+| `alerting.email.username`               | Username of the SMTP server used to send the alert. If empty, uses `alerting.email.from`.  | `""`          |
+| `alerting.email.password`               | Password of the SMTP server used to send the alert                                         | Required `""` |
+| `alerting.email.host`                   | Host of the mail server (e.g. `smtp.gmail.com`)                                            | Required `""` |
+| `alerting.email.port`                   | Port the mail server is listening to (e.g. `587`)                                          | Required `0`  |
+| `alerting.email.to`                     | Email(s) to send the alerts to                                                             | Required `""` |
+| `alerting.email.disable-authentictaion` | Disables authentication. Useful for SMTP relays using IP-based authentication              | `false`       |
```
Comment (TwiN) [inline README.md:480]:
```suggestion
| `alerting.email.disable-authentication` | Disables authentication. Useful for SMTP relays using IP-based authentication              | `false`       |
```

Comment (TwiN) [reply README.md:480]: Is this really necessary though? Could we not just check if `alerting.email.password` is blank, and if it is, assume that the authn is disabled?

**Review by TwiN — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (calvinhenderson) [reply README.md:480]: Yep, I suggested that as an alternative in #607. I just wasn’t sure if that should be the default behavior. If you’re okay with that being the case then I can change it!

File: alerting/provider/email/email.go [hunk 2]
```diff
@@ -66,7 +66,22 @@ func (provider *AlertProvider) Send(endpoint *core.Endpoint, alert *alert.Alert,
...
+
+	var d *gomail.Dialer
+	if len(provider.Password) == 0 {
+		// Get the domain in the From address
+		localName := "localhost"
+		fromParts := strings.Split(provider.From, `@`)
+		if len(fromParts) == 2 {
+			localName = fromParts[1]
+		}
+
```
Comment (TwiN) [inline alerting/provider/email/email.go:78]:
```suggestion
	var d *gomail.Dialer
	if len(provider.Password) == 0 {
		// Get the domain in the From address
		localName := "localhost"
		fromParts := strings.Split(provider.From, `@`)
		if len(fromParts) == 2 {
			localName = fromParts[1]
		}
```

File: alerting/provider/email/email.go [hunk 3]
```diff
@@ -14,12 +14,12 @@ import (
...
-	Password string `yaml:"password"`
-	Host     string `yaml:"host"`
-	Port     int    `yaml:"port"`
-	To       string `yaml:"to"`
+	From        string `yaml:"from"`
+	Username    string `yaml:"username"`
+	Password    string `yaml:"password"`
+	Host        string `yaml:"host"`
+	Port        int    `yaml:"port"`
+	To          string `yaml:"to"`
```
Comment (TwiN) [inline alerting/provider/email/email.go:22]:
```suggestion
	From     string `yaml:"from"`
	Username string `yaml:"username"`
	Password string `yaml:"password"`
	Host     string `yaml:"host"`
	Port     int    `yaml:"port"`
	To       string `yaml:"to"`
```

**Comment by TwiN** [comment]
Excellent work @calvinhenderson, thank you for contributing!

**Pull request merged**